src/libraries/System.Private.CoreLib/src/System/Enum.cs

```diff
@@ -598,391 +804,342 @@ private static bool TryParse(Type enumType, ReadOnlySpan<char> value, bool ignor
             {
                 if (throwOnFailure)
                 {
-                    throw new ArgumentException(SR.Arg_MustContainEnumInfo, nameof(value));
+                    ThrowInvalidEmptyParseArgument();
                 }
+
                 result = null;
                 return false;
             }
 
-            int intResult;
-            uint uintResult;
             bool parsed;
 
-            switch (Type.GetTypeCode(rt))
+            switch (Type.GetTypeCode(rt.GetEnumUnderlyingType()))
```

Comment:
Using the underlyingType with GetTypeCode isn't needed, it automatically uses it with enums.